### PR TITLE
kvm_modb: rewrite kvm to only read/update from modbs

### PR DESCRIPTION
**Pull request by icehess**
Please review but not merge yet!

### CODE REVIEW

File: applications/crossbar/src/modules_v2/cb_vmboxes_v2.erl [hunk 1]
```diff
@@ -576,83 +601,129 @@ load_vmbox(DocId, Context) ->
...
+    end.
+
+-spec fix_envelope(cb_context:context()) -> cb_context:context().
+fix_envelope(Context) ->
+    UpdatedEnvelope =
+        case {cb_context:doc(Context)
+             ,cb_context:resp_envelope(Context)
+             }
+        of
+            {[], Envelope} ->
```
Comment (fenollp) [inline applications/crossbar/src/modules_v2/cb_vmboxes_v2.erl]: Can a cb_context:doc equal `[]`?

File: core/kazoo_voicemail/src/kvm_messages.erl [hunk 2]
```diff
@@ -152,172 +106,103 @@ count_by_owner(AccountId, OwnerId) ->
...
+
+-spec do_update(ne_binary(), kz_json:objects(), update_funs(), kz_json:objects()) -> dict:dict().
+do_update(AccountId, SucceededJObjs, Funs, FailedJObjs) ->
+    ToUpdateDict = apply_fun_and_map_msgs_to_modb(AccountId, SucceededJObjs, Funs, dict:new()),
+    ResultsDict = dict:from_list([{<<"failed">>, FailedJObjs}]),
+    Fun = fun(Db, Js, ResDict) ->
+                  case kz_datamgr:save_docs(Db, Js) of
+                      {'ok', Saved} ->
+                          normalize_bulk_results('undefined', Saved, ResDict);
+                      {'error', R} ->
```
Comment (fenollp) [inline core/kazoo_voicemail/src/kvm_messages.erl]: How about logging this error?

File: core/kazoo_voicemail/src/kz_voicemail.hrl [hunk 3]
```diff
@@ -12,21 +12,22 @@
...
-       ,?SECONDS_IN_DAY * Duration + ?SECONDS_IN_HOUR
+-define(RETENTION_SECONDS(Days),
+        ?SECONDS_IN_DAY * Days + ?SECONDS_IN_HOUR
        ).
 
--record(bulk_res, {succeeded = []  :: ne_binaries()
-                  ,failed = [] :: kz_json:objects()
-                  ,moved = [] :: ne_binaries()
-                  }).
+-define(CHANGE_VMBOX_FUNS(AccountId, NewBoxId, NBoxJ, OldBoxId),
```
Comment (fenollp) [inline core/kazoo_voicemail/src/kz_voicemail.hrl]: Why is this not a function?

File: applications/crossbar/src/modules_v2/cb_vmboxes_v2.erl (hunk 1 above)
Comment (icehess) [inline applications/crossbar/src/modules_v2/cb_vmboxes_v2.erl]: it's contains view results, hence `kz_json:objects()`

**Comment by fenollp** [comment]
@icehess needs rebase btw

**Comment by icehess** [comment]
@fenollp or @jamesaimonetti for logging failed open/save, what should be log level? debug/warning or info?

File: core/kazoo_voicemail/src/kvm_maintenance.erl [hunk 4]
```diff
@@ -65,22 +64,22 @@ cleanup_heard_voicemail(AccountId) ->
     end.
 
 -spec cleanup_heard_voicemail(ne_binary(), pos_integer(), kz_proplist()) -> 'ok'.
-cleanup_heard_voicemail(AccountId, Timestamp, Boxes) ->
-    F = fun (Box) -> cleanup_voicemail_box(AccountId, Timestamp, Box) end,
-    lists:foreach(F, Boxes).
+cleanup_heard_voicemail(AccountId, RetentionSeconds, Boxes) ->
+    _ = [cleanup_voicemail_box(AccountId, RetentionSeconds, Box) || Box <- Boxes],
```
Comment (fenollp) [inline core/kazoo_voicemail/src/kvm_maintenance.erl]: I liked that lists:foreach :(

**Comment by icehess** [comment]
@fenollp why did you merge this? It's still needs some changes/fixes.

**Comment by lazedo** [comment]
@icehess well, it looked ok. if the pr is not to be merged, please make a note of that.

**Comment by icehess** [comment]
@lazedo monster-ui still is unhappy with some of the vm api responses. And I mentioned to do not merge in the first comment :)

**Comment by lazedo** [comment]
@icehess you're actually did, sorry :) thought it was the other that was closed yesterday

**Pull request merged**